Base/res/js/Spreadsheet/runtime.js

```diff
@@ -1,8 +1,79 @@
+// FIXME: Figure out a way to document non-function entities too.
+class Position {
+    constructor(column, row, sheet) {
+        this.column = column;
+        this.row = row;
+        this.sheet = sheet ?? thisSheet;
+        this.name = column + "" + row;
+    }
+
+    static from_name(name) {
+        let sheet = thisSheet;
+        let obj = sheet.parse_cell_name(name);
+        return new Position(obj.column, obj.row, sheet);
+    }
+
+    up(how_many) {
+        how_many = how_many ?? 1;
+        const row = Math.max(0, this.row - how_many);
+        return new Position(this.column, row, this.sheet);
+    }
+
+    down(how_many) {
+        how_many = how_many ?? 1;
+        const row = Math.max(0, this.row + how_many);
+        return new Position(this.column, row, this.sheet);
+    }
+
+    left(how_many) {
+        how_many = how_many ?? 1;
+        const column = Math.min(
+            "Z".charCodeAt(0),
+            Math.max("A".charCodeAt(0), this.column.charCodeAt(0) - how_many)
+        );
+        return new Position(String.fromCharCode(column), this.row, this.sheet);
+    }
+
+    right(how_many) {
+        how_many = how_many ?? 1;
+        const column = Math.min(
+            "Z".charCodeAt(0),
+            Math.max("A".charCodeAt(0), this.column.charCodeAt(0) + how_many)
+        );
+        return new Position(String.fromCharCode(column), this.row, this.sheet);
+    }
+
+    with_column(value) {
+        return new Position(value, this.row, this.sheet);
+    }
+
+    with_row(value) {
+        return new Position(this.column, value, this.sheet);
+    }
+
+    in_sheet(the_sheet) {
+        return new Position(this.column, this.row, sheet(the_sheet));
+    }
+
+    value() {
+        const name = this.column + "" + this.row;
+        return this.sheet[name];
+    }
+
+    valueOf() {
+        return value();
+    }
+
+    toString() {
+        return `<Cell at ${this.name}>`;
+    }
+}
+
 function range(start, end, columnStep, rowStep) {
     columnStep = integer(columnStep ?? 1);
     rowStep = integer(rowStep ?? 1);
-    start = parse_cell_name(start) ?? { column: "A", row: 0 };
-    end = parse_cell_name(end) ?? start;
+    if (!(start instanceof Position)) start = parse_cell_name(start) ?? { column: "A", row: 0 };
+    if (!(end instanceof Position)) end = parse_cell_name(end) ?? start;
```

Comment:
Thanks for using prettier lol, but I hate that it does this. Feel free to wrap the if body in {} to get a line break

Reply:
me two lol